### PR TITLE
Neumann BC ctests

### PR DESCRIPTION
**Pull request by endJunction**
Based on the general approach to include lower dimensional elements presented in ufz/ogs#702 this adds the neumann boundary conditition tests.

 - Provide analytical solutions and add tests for the Neumann bc ctests. Also see the documentation at opengeosys.org/docs.
 - Fixes ufz/ogs#697.

Know issues:
 - I had to set the comparison error to 1e-2, which is pretty large for the analytical benchmarks. Possible resolution are
   1. to provide a sharper margin for each test (probably based on the mesh resolution) instead of general margin for the vtkdiff tests. (ufz/ogs#704)
   2. forward configuration of linear equation system solvers to the actual solvers instead of using the only hardcoded set. (ufz/ogs#705)

### CODE REVIEW

File: Applications/CLI/Tests.cmake [hunk 1]
```diff
@@ -21,6 +21,8 @@ foreach(mesh_size 1e0 1e1 1e2 1e3)
 		EXECUTABLE ogs
 		EXECUTABLE_ARGS cube_${mesh_size}_neumann.prj
 		WRAPPER time
+		TESTER vtkdiff
+        DIFF_DATA cube_${mesh_size}_neumann.vtu D1_left_front_N1_right Result
```
Comment (TomFischer) [inline Applications/CLI/Tests.cmake]: Is the indentation correct?

**Comment by TomFischer** [comment]
Apart from small comment and my comments in Noris PR :fast_forward:

**Comment by bilke** [comment]
Needs to be rebased.

**Comment by TomFischer** [comment]
Rebased.

**Pull request merged**